### DISCUSSION TITLE
Just a question about tauri

### DISCUSSION THREAD

**Discussion by digitalit** [tkaelbel/timekeeping#30] [Q&A]
Hi,

Nice repo. I just want to ask you about tauri and why you use it instead of the Quasar's native Electron way?

I love new stuff (for me) and i was hoping this would be a easyer way of creating an desktop version of my app.

I find Electron and all local setting up quite tricky.

Have a nice day.

Best regards
/Peter Sweden

**Comment by tkaelbel** [comment]
Hi Peter,
thanks for your appreciation!

I heard about Tauri last year and liked to try it out. It has some advantages over Electron.

- It's more efficient in the resource management (The application currently has about 90MB RAM usage and mostly 0,1% CPU usage)
- The built executables are more leight weight (As far as I know in Electron everything is bundled in one big executable. Tauri uses the different OS implementations like on Windows WebView2 to run applications)
- So far Tauri has less security issues than Electron

As you mentioned Quasar does a lot for you about the configuration with Electron. But for me, there is also a lot of overhead produced by Quasar. I like the framework as an UI framework, but want to handle the build stuff by myself.

So far I'm very happy with Tauri. Probably there are some edge cases, where Electron is better, because it exists longer. 

Also there is a Github Action from the Tauri team, which allows you to easily build your application for the different OS.

I hope I could answer your questions. Don't be shy to ask more ;)

Best regards
Tim (Germany)

**Comment by digitalit** [comment]
Danke schön Tim,

So, i'm on a Mac and i wonder if i can create for Windows directly with Tauri or must i like with Electron be on a Windows machine?

I use SSR+PWA and Vite for my frontend (Quasar V.2+) and Adonis V.5 for my backend api. Some say: don't make desktop apps that rely on a backend. I see you have data handled in a local json file and i guess that makes it "backend free"?

So, what is you opinion on desktop apps that rely on a backend api?

Have a nice day.

**Comment by tkaelbel** [comment]
Det var så lite så Peter, (I hope Deepl is correct :D)

If you build it locally, it will only be build for your running OS. But as I mentioned you can easily setup a Github Action to build a production version of your application for linux, mac and windows. Have a look here -> https://github.com/tauri-apps/tauri-action. And as you can see here in the project, we have different installers around -> https://github.com/tkaelbel/timekeeping/releases/tag/v1.0.2

Depends highly on your usecase. For example Spotfiy has also an electron application around. If they would not use a backend, It would probably hard to stream the music.. 
For the timekeeper application, it was totally fine to use a json and store it right on the machine. But as I mentioned, depends highly on your usecase. We want to change the timekeeper to save the data in a sqlite file locally.

I did not try out yet, how Tauri behaves with a backend communication. But I guess it's just a call of a webservice in JS/TS with await and maybe axios. Probably you need to set some permissions for the app, but I'm not sure. You can have a look at the examples from Tauri. Maybe there is something for you -> https://github.com/tauri-apps/tauri/tree/dev/examples

Thanks, have a nice day too!
